pipeline/frontend/metadata/environment.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2023 Woodpecker Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metadata
+
+import (
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var pullRegexp = regexp.MustCompile(`\d+`)
+
+// Environ returns the metadata as a map of environment variables.
+func (m *Metadata) Environ() map[string]string {
+	var (
+		repoOwner    string
+		repoName     string
+		sourceBranch string
+		targetBranch string
+	)
+
+	repoParts := strings.Split(m.Repo.Name, "/")
+	if len(repoParts) == 2 {
+		repoOwner = repoParts[0]
+		repoName = repoParts[1]
+	} else {
+		repoName = m.Repo.Name
+	}
+
+	branchParts := strings.Split(m.Curr.Commit.Refspec, ":")
+	if len(branchParts) == 2 {
+		sourceBranch = branchParts[0]
+		targetBranch = branchParts[1]
+	}
+
+	params := map[string]string{
+		"CI":                     m.Sys.Name,
+		"CI_REPO":                m.Repo.Name,
+		"CI_REPO_OWNER":          repoOwner,
+		"CI_REPO_NAME":           repoName,
+		"CI_REPO_SCM":            "git",
+		"CI_REPO_URL":            m.Repo.Link,
+		"CI_REPO_CLONE_URL":      m.Repo.CloneURL,
+		"CI_REPO_DEFAULT_BRANCH": m.Repo.Branch,
+		"CI_REPO_PRIVATE":        strconv.FormatBool(m.Repo.Private),
+		"CI_REPO_TRUSTED":        "false", // TODO should this be added?
```

Comment:
Is this ever set at some other place otherwise I would think about just removing it as its never set anyways.

Reply:
I'll add this info as it's usefull for plugins like buildx